main.py

```diff
@@ -43,34 +53,45 @@ def apply_for_single_session(
         By.XPATH, "//input[@type='text' and @placeholder='シリアルナンバー']"
     )
 
-    print("Filling code")
+    logging.info("<Status> Filling code")
     serial_code_box.send_keys(code)
 
     apply_button = driver.find_element(By.XPATH, "//button[text()='お申込みへ']")
     apply_button.click()
 
-    # Attempt of an exception handling method aimed to react when the code is either wrong or used.
-    time.sleep(3)
-    error_messages = driver.find_elements(By.CLASS_NAME, "varidation")
-    # print(len(error_messages))
-    if error_messages:
-        error_msg = error_messages[1].text
-        if error_msg == "利用回数を超えたためお申込みできません。":
-            print(f"Code {code} has been used for {session_name} before. Please check again.")
-        elif error_msg == "申し込み情報が正しくありません。":
-            print(f"Code {code} is incorrect. Please check again.")
-        return False
-
-    # Waiting until the new page loads
+    # Waiting until the new page loads or an error message pops out
     WebDriverWait(driver, 10).until(
-        EC.element_to_be_clickable(
-            (
-                By.XPATH,
-                "//button[text()='申込む' and @data-title='★ 必ずお読みください ★']",
+        EC.any_of(
+            EC.element_to_be_clickable(
+                (
+                    By.XPATH,
+                    "//button[text()='申込む' and @data-title='★ 必ずお読みください ★']",
+                )
+            ),
+            EC.presence_of_element_located(
+                (
+                    By.XPATH,
+                    "//div[@name='ninsho_key_whole_error_info']/p[contains(text(),'ません')]"
+                )
             )
-        ),
+        )
     )
 
+    # Figure out if there is an error message
+    try:
+        error_message = driver.find_element(
+            By.XPATH,
+            "//div[@name='ninsho_key_whole_error_info']/p[contains(text(),'ません')]"
+        )
+        if error_message:
+            error_msg = error_message.text
+            if error_msg == "利用回数を超えたためお申込みできません。":
+                return Status.USED_CODE
+            elif error_msg == "申し込み情報が正しくありません。":
+                return Status.INVALID_CODE
+    except:
```

Comment:
```suggestion
    except Exception:
```

https://docs.astral.sh/ruff/rules/bare-except/